staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -500,30 +515,13 @@ func (p *patcher) applyPatch(_ request.Context, _, currentObject runtime.Object)
 		currentResourceVersion = currentMetadata.GetResourceVersion()
 	}
 
-	switch {
-	case p.originalObjJS == nil && p.originalObjMap == nil:
```

Comment:
it's vaguely unsettling to not know whether the initial state the subsequent attempts rely on was actually set up correctly. if `firstPatchAttempt` fails, should we ever try again?

Reply:
Ah, is that why it is like it is? Let me think about that.